src/main.rs

```diff
@@ -135,34 +136,59 @@ async fn insert_access_keys_from_dumped_state(
 
 async fn insert_receipts(
     height: near_indexer::near_primitives::types::BlockHeight,
-    chunks: &[near_indexer::near_primitives::views::ChunkView],
+    chunks: &[near_indexer::IndexerChunkView],
+    local_receipts: &[near_indexer::near_primitives::views::ReceiptView],
     pool: &Pool<ConnectionManager<PgConnection>>,
 ) {
-    let access_keys = chunks
+    let mut access_keys: Vec<AccessKey> = local_receipts
         .iter()
-        .map(|chunk| &chunk.receipts)
-        .flatten()
         .filter_map(|receipt| match receipt.receipt {
             near_indexer::near_primitives::views::ReceiptEnumView::Action { .. } => {
-                Some(AccessKey::from_receipt_view(receipt, height))
+                Some(AccessKey::from_receipt_view(&receipt, height))
```

Comment:
Given you use `filter_map + flatten`, you can use `flat_map` and return an empty vec when the receipt type is not Action.